### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by algerina**
In this milestone I created:

-  Model classes for all entities.
- Set up associations between models.
- A method that returns the 3 most recent posts for a given user.
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
- A method that updates the comments counter for a post.
- A method called current_user in your ApplicationController.
-

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.length)
```
Comment (dasileker) [inline app/models/like.rb]: - [x] Nice job on implementing the nesseccary codes, however,  the update method updates the posts or comments, and like each time you push something new to your database and this is good but you can use increments instead to push new posts or comments or like with ease.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @algerina,** 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights

- The PR is very detailed 👌.
- The readme file is very detailed 👌.
- The coding is very clean 👌.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (algerina) [reply app/models/like.rb]: Thank you for your valuable feedback.

**Review by julie-ify — APPROVED** [review]
Hi @algerina, 

# Status: Approved ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**